heron/instance/src/java/com/twitter/heron/instance/bolt/BoltInstance.java

```diff
@@ -192,8 +192,8 @@ public void readTuplesAndExecute(Communicator<HeronTuples.HeronTupleSet> inQueue
       int nValues = topologyContext.getComponentOutputFields(
           stream.getComponentName(), stream.getId()).size();
 
-      // We would reuse the System.nanoTime()
-      long currentTime = startOfCycle;
+      // Record current time at the start of processing a set of tuples
+      long currentTime = System.nanoTime();
```

Comment:
There was an intent optimization to reduce one  `System.nanoTime()` call per tuple execution. In the new commit this optimization is removed to simplify the reasoning. 
Now it is much easier to understand.